lib/checkother.cpp

```diff
@@ -1640,6 +1640,10 @@ void CheckOther::checkConstPointer()
                         if (!isVariableChangedByFunctionCall(ftok, vt->pointer, var->declarationId(), mSettings, &inconclusive) && !inconclusive)
                             continue;
                     }
+                } else if (!parent->isCast()) {
+                    const auto dir = mSettings->library.getArgDirection(ftok, argn + 1);
+                    if (dir == Library::ArgumentChecks::Direction::DIR_IN)
```

Comment:
It seems like this logic should be in `isVariableChangedByFunctionCall`, but I wonder why we skip it when the parent is a cast.

Reply:
For this test:
~~~
void f(int, const int*);
void f(int i, int* p) {
    f(i, const_cast<const int*>(p));
}
~~~
and also `testMisusedScopeObjectNamespace()`. Seems not ideal though.

Reply:
`isVariableChangedByFunctionCall()` already has handling for library functions, but it is not quite aligned with `getTokenArgumentFunction()` (constructors, init lists).

Reply:
> isVariableChangedByFunctionCall() already has handling for library functions, but it is not quite aligned with getTokenArgumentFunction() (constructors, init lists).

We should probably have an issue to track that.

Reply:
If we only rely on `isVariableChangedByFunctionCall()`, there would be FPs for those:
~~~
struct A {
    int* x;
    A(int* y) : x(y) {}
};
void f(int* q) {
    int* a[1] = { q };
}
~~~
I'm not convinced that `isVariableChangedByFunctionCall()` should return `true` here.